File: pkg/model/host/host.go
package host

import (
	"github.com/KubeOperator/KubeOperator/pkg/model/common"
	"github.com/KubeOperator/KubeOperator/pkg/model/credential"
	uuid "github.com/satori/go.uuid"
)

const (
	Running      string = "Running"
	Waring       string = "Warning"
	Disconnect   string = "DisConnect"
	SshError     string = "SshError"
	AnsibleError string = "AnsibleError"
)

type Host struct {
	common.BaseModel
	Credential   credential.Credential
	ID           string
	Name         string `gorm:"not null;unique"`
	Memory       int
	CpuCore      int
	Os           string
	OsVersion    string
	GpuNum       int
	GpuInfo      string
	Ip           string
	Port         int
	CredentialID string
	Status       string
	NodeID       string
	Volumes      []Volume
}

type Volume struct {
	common.BaseModel
	size string
	name string
}

func (h *Host) BeforeCreate() (err error) {
	h.ID = uuid.NewV4().String()
	return err
}

func (h Host) TableName() string {
	return "ko_host"
}

File: ui/src/app/business/host/host.ts
import {BaseModel, BaseRequest} from '../../shared/class/BaseModel';

export class Host extends BaseModel {
    name: string;
    ip: string;
    port: string;
    os: string;
    osVersion: string;
    memory: string;
    cpuCore: number;
    gpuNum: number;
    gpuInfo: string;
    status: string;
    volumes: Volume[];
    projectName: string;
    clusterName: string;
    clusterId: string;
    zoneName: string;
    message: string;
    hasGpu: boolean;
    architecture: string;
}

export class Volume extends BaseModel {
    size: string;
    name: string;
    hostId: string;
}

export class HostCreateRequest extends BaseRequest {
    name: string;
    ip: string;
    port: string;
    credentialId: string;
}

export class HostSync {
    hostName: string;
    hostStatus: string;
}

export class Project {
    id: string;
    name: string;
}
